torch/csrc/jit/serialization/import_source.cpp

```diff
@@ -639,23 +639,34 @@ struct SourceImporterImpl : public Resolver,
       const QualifiedName& qualified_name,
       const ClassDef& named_tuple_def) {
     ScriptTypeParser type_parser(shared_from_this());
-    std::vector<std::string> field_names;
+    std::vector<std::pair<std::string, IValue>> fields;
     std::vector<TypePtr> field_types;
     for (const auto& statement : named_tuple_def.body()) {
       if (statement.kind() != TK_ASSIGN) {
         throw ErrorReport(statement.range())
             << "Unexpected statement in NamedTuple body: "
                "only attribute annotations are currently supported.";
       }
-
       const auto assign = Assign(statement);
-      auto name = Var(assign.lhs()).name().name();
-      field_names.emplace_back(std::move(name));
+
+      auto name = Var(Assign(statement).lhs()).name().name();
+      auto default_val = IValue();
+      if (assign.rhs().present()) {
+        std::vector<IValue> parsed = type_parser.evaluateDefaults(
+            assign.range(), {assign.rhs().get()}, {assign.type().get()});
```

Comment:
What's the rationale behind using `assign.range()` instead of `assign.rhs().range()`?

Reply:
fixed